socs/agents/generator/agent.py

```diff
@@ -99,6 +99,46 @@ def read(self, client):
         return return_data
 
 
+class Generator:
+    """Functions to communite with the Generator controller
+
+    Attributes
+    ----------
+    read_strings : list
+        Strings of registers defined in config to read from
+    """
+
+    def __init__(self):
+        self.read_strings = []
+
+        read_config = os.path.join(os.path.dirname(__file__), "config.yaml")
+        with open(read_config) as f:
+            data = yaml.load(f, Loader=yaml.SafeLoader)
+        self.build_config([data])
+
+    def init_client(self, host, port):
+        client = ModbusClient(host, port, auto_open=True, auto_close=False)
+        return client
```

Comment:
Great, I made one small adjustment here: https://github.com/simonsobs/socs/pull/436/commits/6130bd03448c05570c5f99165c769d47d61b5a77

Since you have `auto_open=True` when making the ModbusClient, the TCP connection should be open. Calling `client.open()` again will close and reopen the TCP connection. This patch just uses the `client.is_open` check that you use below.